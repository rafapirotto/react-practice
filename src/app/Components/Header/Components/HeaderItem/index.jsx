import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const propTypes = {
  title: PropTypes.string.isRequired,
  styles: PropTypes.string,
};

const defaultProps = {
  styles: '',
};

const HeaderItem = ({ title, styles }) => (
  <div className={styles}>
    <Link to="/">{title}</Link>
  </div>
);

HeaderItem.propTypes = propTypes;
HeaderItem.defaultProps = defaultProps;

export default HeaderItem;
