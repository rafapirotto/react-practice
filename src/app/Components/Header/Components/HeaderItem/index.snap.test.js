import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';

import HeaderItem from '.';

it('matches the snapshot', () => {
  const component = renderer.create(
    <Router>
      <HeaderItem title="title" styles="" to="/" />
    </Router>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
