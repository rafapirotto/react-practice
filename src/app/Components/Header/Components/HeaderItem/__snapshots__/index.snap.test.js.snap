// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It matches the snapshot 1`] = `
<div
  className=""
>
  <a
    href="/"
    onClick={[Function]}
  >
    title
  </a>
</div>
`;
